addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1334,77 +1430,90 @@ var SnippetsMenu = Widget.extend({
      */
-    _activateSnippet: async function ($snippet, previewMode, ifInactiveOptions) {
+    _activateSnippet: async function ($snippetBlock, previewMode, ifInactiveOptions) {
         if (this._blockPreviewOverlays && previewMode) {
             return;
         }
-        if ($snippet && !$snippet.is(':visible')) {
+        if (!$snippetBlock.is(':visible')) {
             return;
         }
+
         const exec = previewMode
             ? action => this._mutex.exec(action)
             : action => this._execWithLoadingEffect(action, false);
-        return exec(() => {
-            return new Promise(resolve => {
-                // Take the first parent of the provided DOM (or itself) which
-                // should have an associated snippet editor and create + enable it.
-                if ($snippet && $snippet.length) {
-                    $snippet = globalSelector.closest($snippet);
-                    if ($snippet.length) {
-                        return this._createSnippetEditor($snippet).then(resolve);
-                    }
-                }
-                resolve(null);
-            }).then(editorToEnable => {
-                if (ifInactiveOptions && this._enabledEditorHierarchy.includes(editorToEnable)) {
-                    return editorToEnable;
-                }
 
-                const editorToEnableHierarchy = [];
-                let current = editorToEnable;
-                while (current && current.$target) {
-                    editorToEnableHierarchy.push(current);
-                    current = current.getParent();
+        return exec(async () => {
+            let snippetEditor;
+            // Take the first parent of the provided DOM (or itself) which
+            // should have an associated snippet editor and create + enable it.
+            if ($snippetBlock.length) {
+                const $snippet = globalSelector.closest($snippetBlock);
+                if ($snippet.length) {
+                    snippetEditor = await this._getSnippetEditor($snippet);
                 }
+            }
+            if (ifInactiveOptions && this._enabledSnippetEditorsHierarchy.includes(snippetEditor)) {
+                return snippetEditor;
+            }
 
-                // First disable all editors...
-                for (let i = this.snippetEditors.length; i--;) {
-                    const editor = this.snippetEditors[i];
-                    editor.toggleOverlay(false, previewMode);
-                    if (!previewMode && !editorToEnableHierarchy.includes(editor)) {
-                        editor.toggleOptions(false);
-                    }
-                }
-                // ... if no editors are to be enabled, look if any have been
-                // enabled previously by a click
-                if (!editorToEnable) {
-                     editorToEnable = this.snippetEditors.find(editor => editor.isSticky());
-                     previewMode = false;
-                }
```

Comment:
@Goaman Any idea why this was removed?

Reply:
I don't remember any case for this removal.